### PR TITLE
[PyTorch] Remove Megatron-LM convergence test

### PR DESCRIPTION
**Pull request by timmoon10**
# Description

The L3 convergence test is more of a run script than an actual test and it just outputs logs that need to be manually inspected for problems. It also uses an idiosyncratic dataset. In the future we should think of a better approach for validating convergence, perhaps using an MLPerf dataset and actually failing if the learning quality or performance fall outside of some hard-coded bounds.

## Type of change

- [ ] Documentation change (change only to the documentation, either a fix or a new content)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Infra/Build change
- [ ] Code refactoring
- [x] Testing

## Changes

- Remove Megatron-LM convergence test

# Checklist:

- [x] I have read and followed the [contributing guidelines](https://github.com/NVIDIA/TransformerEngine/blob/main/CONTRIBUTING.rst)
- [ x The functionality is complete
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by timmoon10** [comment]
Pipeline 24689192

**Comment by ksivaman** [comment]
I think it's good to keep these tests/files until we have a concrete replacement ready. I agree that we should run these more often and have an automatic convergence check for the metrics, maybe val loss/ppl?

File: qa/L3_pytorch_convergence_test/test.sh [hunk 1]
```diff
@@ -1,14 +0,0 @@
...
-set -e
-
-: ${TE_PATH:=/opt/transformerengine}
-
-pip install prettytable
-git clone https://github.com/NVIDIA/Megatron-LM.git
-cd Megatron-LM
-git checkout b3375a0e38c10e2300ef4be031f7dcabab52b448
-pytest -v -s $TE_PATH/tests/pytorch/distributed/test_convergence.py
-python $TE_PATH/tests/pytorch/distributed/print_logs.py
```
Comment (ksivaman) [inline qa/L3_pytorch_convergence_test/test.sh:14]: We could modify this to not just print but have a threshold for validation PPL. Currently the test is only functional aside from the manual checking of logs.
